S2739: parse JSON tuple input,

hjeeves opened the pull request:
work covers these tasks: ta13810, ta13717, ta 13579: implement tuple classes, parse those sanely throughout the code, handle multipart data using SyncInput. 

Integration tests for JSON data added in wwebui/downloadManager, s2739a branch - this needs to be code reviewed along with code here in opencadc/apps.

pdowler reviewed (changes requested):
DownloadTuple, ParsedURI: formatting is not readable.

cadc-download-manager: strange version jump - always check the actual version change(s) in build file.

cadc-upload-manager: why any changes here?

hjeeves commented:
> DownloadTuple, ParsedURI: formatting is not readable.
> 
> cadc-download-manager: strange version jump - always check the actual version change(s) in build file.
> 
> cadc-upload-manager: why any changes here?

- DownloadTuple & ParsedURI format should be fixed now
- downgraded cadc-download-manager build version
- only thing I can imagine happened to cadc-upload-manager is those symlinks got butchered when I copied the directory to one of a different name to preserver work as I switched between s2739 branch & s2739a branch. Reverted to original state now.

pdowler reviewed (changes requested):
in JSP pages: https://localhost is never going to work (server certificate won't match hostname). Can this skinning stuff be dropped entirely?

some details below

hjeeves commented on cadc-download-manager/src/main/java/ca/nrc/cadc/dlm/DownloadTuple.java:
```diff
@@ -0,0 +1,51 @@
+
+    //    tupleID (URI) will be translated into URI when needed
+    //    this format required for use in DownloadIterators
+    public final String tupleID;
```
I'd left it as String because in the scope of the current code the value was never used as a URI. Changed it over to provide some typesafe-ness.

hjeeves commented on cadc-download-manager/src/main/java/ca/nrc/cadc/dlm/DownloadTuple.java:
```diff
@@ -0,0 +1,51 @@
+    // unsure we need Region or not - only interesting
+    // in cadc-download-manager when DataLinkClient is putting
+    // together cutouts
+    private Region shape;
```
It's still not clear to me whether this value is needed or not. Unless there's some validation that will happen prior to DataLinkClient making calls (where it didn't before)?

hjeeves replied:
changed to be 'Shape' and have the cadc-dali dependency

hjeeves commented on cadc-download-manager/src/main/java/ca/nrc/cadc/dlm/ParsedURI.java:
```diff
@@ -0,0 +1,36 @@
+package ca.nrc.cadc.dlm;
```
Ok - ParsedURI Throwable attribute merged into DownloadTuple, DownloadDescriptor instantiation in DownloadUtil now uses DownloadTuple instance.

hjeeves commented on cadc-download-manager-server/build.gradle:
```diff
@@ -16,7 +16,7 @@ repositories {
 }
 
-version = '1.3.0'
+version = '1.3.1'
```
ok

hjeeves commented on cadc-download-manager/src/main/java/ca/nrc/cadc/dlm/DownloadTuple.java:
```diff
@@ -0,0 +1,51 @@
+package ca.nrc.cadc.dlm;
```
ok

hjeeves replied:
still have to remove skinning from .jsps.

hjeeves commented on cadc-download-manager-server/src/test/java/ca/nrc/cadc/dlm/server/DLMInputHandlerTest.java:
```diff
@@ -0,0 +1,124 @@
+    private static String LABEL_STR = "label";
+
+    // Using Badgerfish json so that JsonInputter can read it correctly
+    private static String TUPLE_JSON_URI = "{\"tupleID\":\"{\"$\":\"" + URI_STR + "\"}}";
```
added comments for now, removed unused JSON from tests so overall it should be more readable.

hjeeves commented:
Data model needs to be re-worked, too many changes to keep up with in this pr. Closing in favour of xp work in my branch.
